src/main/java/nwoolcan/model/brewery/warehouse/stock/StockImpl.java

```diff
@@ -111,7 +111,9 @@ public final Date getLastChangeDate() {
                             .peek(q ->
```

Comment:
shouldn't it be a flatmap since `Quantities.add(...)` returns a `Result<Quantity>`?

Reply:
That operation does not have possibility to fail because everything is already checked. In that case I am not mapping: I am assigning a value

Reply:
ok i understand, so why check it before when you can use checks in `Quantities`?
also even if you check it before, you still should not call `getValue()` on a result without checking.
still, i'll approve this, but consider what i said